File: generate_favicon.js
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function generateFavicon() {
  const inputPath = path.join(__dirname, 'logo-processed.png');
  const outputIco1 = path.join('c:', 'Users', 'AbinaM', 'Desktop', 'We-brand', 'webrand', 'src', 'app', 'favicon.ico');
  const outputIco2 = path.join(__dirname, 'public', 'favicon.ico');
  const outputSvg2 = path.join(__dirname, 'public', 'favicon.svg');
  const outputWebp = path.join('c:', 'Users', 'AbinaM', 'Desktop', 'We-brand', 'webrand', 'public', 'favicon.webp');

  try {
    const size = 256;
    const padding = 40;
    const logoSize = size - (padding * 2);

    // 1. Resize the logo
    const resizedLogo = await sharp(inputPath)
      .resize(logoSize, logoSize, { fit: 'inside' })
      .toBuffer();

    // 2. Composite over blue circle
    // Creating an SVG circle as the base
    const svgBase = `
      <svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
        <circle cx="${size/2}" cy="${size/2}" r="${size/2}" fill="#2563c9" />
      </svg>
    `;

    const finalImageBuffer = await sharp(Buffer.from(svgBase))
      .composite([{ input: resizedLogo, gravity: 'center' }])
      .png()
      .toBuffer();

    // 3. Save as various formats
    
    // We-brand app router uses favicon.ico in src/app
    await sharp(finalImageBuffer)
      .resize(64, 64)
      .toFile(outputIco1.replace('.ico', '.png')); // Save as PNG first, then we can rename it. 
      // Next.js actually prefers favicon.ico, but it also accepts favicon.png in app router (often as icon.png or icon.ico)
      // Actually, sharp can't output .ico directly in older versions, we can output .png and rename to .ico or just use icon.png
      
    // Let's create an icon.png in src/app for Next.js App Router (it automatically uses icon.png)
    const nextjsIconPng = path.join('c:', 'Users', 'AbinaM', 'Desktop', 'We-brand', 'webrand', 'src', 'app', 'icon.png');
    await sharp(finalImageBuffer).resize(64, 64).png().toFile(nextjsIconPng);
    
    // Replace public/favicon.webp
    await sharp(finalImageBuffer).resize(64, 64).webp().toFile(outputWebp);
    
    // For Payload CMS
    // Overwrite public/favicon.svg with a basic SVG
    const finalSvg = `
      <svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
        <circle cx="${size/2}" cy="${size/2}" r="${size/2}" fill="#2563c9" />
        <image href="data:image/png;base64,${resizedLogo.toString('base64')}" x="${padding}" y="${padding}" width="${logoSize}" height="${logoSize}" />
      </svg>
    `;
    fs.mkdirSync(path.dirname(outputSvg2), { recursive: true });
    fs.writeFileSync(outputSvg2, finalSvg);
    
    console.log("Favicons generated successfully!");

  } catch (error) {
    console.error('Error generating favicon:', error);
  }
}

generateFavicon();

File: generate_favicon.py
from PIL import Image, ImageDraw
import os

input_path = r"c:\Users\AbinaM\Desktop\we-brand-cms\logo-processed.png"
output_ico1 = r"c:\Users\AbinaM\Desktop\We-brand\webrand\src\app\favicon.ico"
output_png1 = r"c:\Users\AbinaM\Desktop\We-brand\webrand\src\app\favicon.png"
output_ico2 = r"c:\Users\AbinaM\Desktop\we-brand-cms\public\favicon.ico"
output_svg2 = r"c:\Users\AbinaM\Desktop\we-brand-cms\public\favicon.svg"

size = (256, 256)
bg_color = "#2563c9"

# Create a new image with transparent background
icon = Image.new("RGBA", size, (255, 255, 255, 0))
draw = ImageDraw.Draw(icon)

# Draw a circle with the brand color
draw.ellipse((0, 0, size[0], size[1]), fill=bg_color)

try:
    # Open the processed logo (which is white)
    logo = Image.open(input_path).convert("RGBA")
    
    # Resize logo to fit inside the circle (give it some padding)
    logo_size = int(size[0] * 0.7)
    logo = logo.resize((logo_size, logo_size), Image.Resampling.LANCZOS)
    
    # Calculate position to center the logo
    pos = ((size[0] - logo_size) // 2, (size[1] - logo_size) // 2)
    
    # Paste the logo using its own alpha as a mask
    icon.paste(logo, pos, mask=logo)

    # Ensure directories exist
    os.makedirs(os.path.dirname(output_ico1), exist_ok=True)
    os.makedirs(os.path.dirname(output_ico2), exist_ok=True)

    # Save as ICO (multiple sizes for good browser support)
    icon.save(output_ico1, format="ICO", sizes=[(16,16), (32,32), (48,48), (64,64)])
    icon.save(output_ico2, format="ICO", sizes=[(16,16), (32,32), (48,48), (64,64)])
    icon.save(output_png1, format="PNG")
    
    # Optionally save a PNG for the SVG to reference, or just overwrite the SVG with an image data URI if possible.
    # We will just write a simple SVG that embeds the base64 PNG or just delete the SVG so the ICO is used.
    
    print("Favicons generated successfully!")
except Exception as e:
    print("Error:", e)
